torch/csrc/jit/argument_spec.h

```diff
@@ -0,0 +1,180 @@
+#include <Python.h>
+#include <iostream>
+#include <vector>
+#include "torch/csrc/autograd/variable.h"
+#include "torch/csrc/utils/hash.h"
+
+namespace torch { namespace jit {
+
+// GraphExecutor creates specializations of Graphs for different dimensionalitities
+// and types of inputs.
+
+// ArgumentSpec represents one particular specialization.
+// It is designed so that it can be created, hashed, and compared quickly
+// since it is used along the hot-path of the JIT to check if the code
+// we have created is valid for the given inputs.
+
+// TensorInfoPOD is only used internally in ArgumentSpec
+// API users should use TensorInfo
+struct TensorInfoPOD {
```

Comment:
Don't you have to add `__attribute__(packed)` to be sure that it really is 64-bit? We're playing with fire in here. Remember that people build PyTorch not only on x86, but also e.g. on ARM, POWER and I don't know what are the alignment restrictions there.

Reply:
Yeah, will do. Probably safer to make it packed, though I can't imagine the case where this wouldn't pack naturally.

Reply:
I tried completely removing the reinterpret_casts, but it turns out to require more major changes. My thought is that we just keep this as is for now, and once we actually can time real things along this pathway, we can make a more informed decision if it is better to {keep, improve, de-optimize} this code.

Reply:
I'm not even sure how would you do that. I'm ok with merging as is, because as I said Python enforces `-fno-strict-aliasing` on us anyway.

Reply:
Well, the `static_assert` will let us know if it doesn't pack :)